### PR TITLE
Add null check before enumerating the enumerable

### LINKED ISSUE #976: Flow Launcher crashes when the computer is not connected to the internet
Hi. I have experienced a bug where Flow launcher crashes with the following error message, every time I try to type something into the flow launcher instance when I am not connected to the internet. It crashes for each letter typed. I had done some testing with different plugin settings try to nail down the issue. I think the issue is the web search plugin's search query autocomplete option that causes flow launcher to crash, as it has no source parameter due to the lack of internet connection. Showing the search result as "not connected to the internet" or not showing a result at all would be a good exception. Thank you!

```
Flow Launcher version: 1.9.1
OS Version: Microsoft Windows NT 10.0.22000.0
IntPtr Length: 8
x64: True


Python Path: 
Date: 01/12/2022 16:52:32
Exception:
System.ArgumentNullException: Value cannot be null. (Parameter 'source')
   at System.Linq.ThrowHelper.ThrowArgumentNullException(ExceptionArgument argument)
   at System.Linq.Enumerable.Any[TSource](IEnumerable`1 source)
   at Flow.Launcher.Plugin.WebSearch.Main.UpdateResultsFromSuggestionAsync(List`1 results, String keyword, String subtitle, SearchSource searchSource, Query query, CancellationToken token) in C:\projects\flow-launcher\Plugins\Flow.Launcher.Plugin.WebSearch\Main.cs:line 116
   at Flow.Launcher.Plugin.WebSearch.Main.QueryAsync(Query query, CancellationToken token) in C:\projects\flow-launcher\Plugins\Flow.Launcher.Plugin.WebSearch\Main.cs:line 92
   at Flow.Launcher.Core.Plugin.PluginManager.<>c__DisplayClass22_0.<<QueryForPluginAsync>b__0>d.MoveNext() in C:\projects\flow-launcher\Flow.Launcher.Core\Plugin\PluginManager.cs:line 192
--- End of stack trace from previous location ---
   at Flow.Launcher.Infrastructure.Stopwatch.DebugAsync(String message, Func`1 action) in C:\projects\flow-launcher\Flow.Launcher.Infrastructure\Stopwatch.cs:line 40
   at Flow.Launcher.Core.Plugin.PluginManager.QueryForPluginAsync(PluginPair pair, Query query, CancellationToken token) in C:\projects\flow-launcher\Flow.Launcher.Core\Plugin\PluginManager.cs:line 211
   at Flow.Launcher.ViewModel.MainViewModel.<>c__DisplayClass137_0.<<QueryResults>g__QueryTask|2>d.MoveNext() in C:\projects\flow-launcher\Flow.Launcher\ViewModel\MainViewModel.cs:line 623
--- End of stack trace from previous location ---
   at Flow.Launcher.ViewModel.MainViewModel.QueryResults() in C:\projects\flow-launcher\Flow.Launcher\ViewModel\MainViewModel.cs:line 624
   at System.Threading.Tasks.Task.<>c.<ThrowAsync>b__140_0(Object state)
   at System.Windows.Threading.ExceptionWrapper.InternalRealCall(Delegate callback, Object args, Int32 numArgs)
   at System.Windows.Threading.ExceptionWrapper.TryCatchWhen(Object source, Delegate callback, Object args, Int32 numArgs, Delegate catchHandler)
```

### PR DESCRIPTION
**Pull request by taooceros**
close #976

### CODE REVIEW

File: Plugins/Flow.Launcher.Plugin.WebSearch/Main.cs [hunk 1]
```diff
@@ -118,32 +117,31 @@ public async Task<List<Result>> QueryAsync(Query query, CancellationToken token)
...
+                SubTitle = subtitle,
+                Score = score,
+                IcoPath = searchSource.IconPath,
+                Action = c =>
                 {
-                    Title = o,
-                    SubTitle = subtitle,
-                    Score = score,
-                    IcoPath = searchSource.IconPath,
-                    ActionKeywordAssigned = searchSource.ActionKeyword == SearchSourceGlobalPluginWildCardSign ? string.Empty : searchSource.ActionKeyword,
```
Comment (jjw24) [inline Plugins/Flow.Launcher.Plugin.WebSearch/Main.cs:136]: why is this removed?

File: Plugins/Flow.Launcher.Plugin.WebSearch/Main.cs [hunk 2]
```diff
@@ -85,8 +85,7 @@ public async Task<List<Result>> QueryAsync(Query query, CancellationToken token)
 
                 ResultsUpdated?.Invoke(this, new ResultUpdatedEventArgs
                 {
-                    Results = results,
-                    Query = query
+                    Results = results, Query = query
```
Comment (jjw24) [inline Plugins/Flow.Launcher.Plugin.WebSearch/Main.cs:88]:
```suggestion
                    Results = results,
                    Query = query
```

File: Plugins/Flow.Launcher.Plugin.WebSearch/Main.cs [hunk 3]
```diff
@@ -167,7 +165,9 @@ void Init()
                 // Custom images directory is in the WebSearch's data location folder 
                 var name = Path.GetFileNameWithoutExtension(_context.CurrentPluginMetadata.ExecuteFileName);
                 CustomImagesDirectory = Path.Combine(DataLocation.PluginSettingsDirectory, name, "CustomIcons");
-            };
+            }
+
+            ;
```
Comment (jjw24) [inline Plugins/Flow.Launcher.Plugin.WebSearch/Main.cs:170]:
```suggestion
            };
```

**Review by jjw24 — CHANGES_REQUESTED** [review]
see comments, version bump too please

File: Plugins/Flow.Launcher.Plugin.WebSearch/Main.cs (hunk 1 above)
Comment (taooceros) [reply Plugins/Flow.Launcher.Plugin.WebSearch/Main.cs:136]: It hasn't been removed, just move to a higher level so displayed like removal, 🤣

Comment (jjw24) [reply Plugins/Flow.Launcher.Plugin.WebSearch/Main.cs:136]: no, this is actually removed, you can do a search, the original file has two of these, this pr file has only one.

File: Plugins/Flow.Launcher.Plugin.WebSearch/Main.cs [hunk 4]
```diff
@@ -118,32 +118,31 @@ public async Task<List<Result>> QueryAsync(Query query, CancellationToken token)
...
+            token.ThrowIfCancellationRequested();
 
-                var resultsFromSuggestion = suggestions?.Select(o => new Result
+            var resultsFromSuggestion = suggestions?.Select(o => new Result
+            {
+                Title = o,
+                SubTitle = subtitle,
+                Score = score,
+                IcoPath = searchSource.IconPath,
+                Action = c =>
```
Comment (jjw24) [inline Plugins/Flow.Launcher.Plugin.WebSearch/Main.cs:138]:
```suggestion
                ActionKeywordAssigned = searchSource.ActionKeyword == SearchSourceGlobalPluginWildCardSign ? string.Empty : searchSource.ActionKeyword,
                Action = c =>
```

**Review by jjw24 — APPROVED** [review]

**Review by jjw24 — APPROVED** [review]
